src/fcio.c

```diff
@@ -449,6 +453,25 @@ Returns the internal FCIOStream object on success or NULL on error.
   return xio;
 }
 
+
+static inline int fcio_put_config(FCIOStream output, fcio_config* config)
+{
+  FCIOWriteMessage(output,FCIOConfig);
+  FCIOWriteInt(output,config->adcs);
+  FCIOWriteInt(output,config->triggers);
+  FCIOWriteInt(output,config->eventsamples);
+  FCIOWriteInt(output,config->blprecision);
+  FCIOWriteInt(output,config->sumlength);
+  FCIOWriteInt(output,config->adcbits);
+  FCIOWriteInt(output,config->mastercards);
+  FCIOWriteInt(output,config->triggercards);
+  FCIOWriteInt(output,config->adccards);
+  FCIOWriteInt(output,config->gps);
+  FCIOWriteInts(output,(config->adcs+config->triggers),config->tracemap);
+
+  return FCIOFlush(output);
+}
+
 /*=== Function ===================================================*/
 
 int FCIOPutConfig(FCIOStream output, FCIOData *input)
```

Comment:
As mentioned above this (and all functions below) needs an additional null-pointer check on `input`.

Reply:
Added